src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -239,12 +239,8 @@ public ExamAccessService(ExamRepository examRepository, ExerciseGroupRepository
         if (courseAndExamAccessFailure.isPresent()) {
             return courseAndExamAccessFailure;
         }
-        Optional<ExerciseGroup> exerciseGroup = exerciseGroupRepository.findById(exerciseGroupId);
-        if (exerciseGroup.isEmpty()) {
-            return Optional.of(notFound());
-        }
-        if (!exerciseGroup.get().getExam().getId().equals(examId)) {
-            return Optional.of(conflict());
+        if (exerciseGroup.getExam() == null || !exerciseGroup.getExam().getId().equals(examId) || !exerciseGroup.getExam().getCourse().getId().equals(courseId)) {
```

Comment:
For the sake of readability, I'd suggest introducing a local Exam variable and replace the three instances of exerciseGroup.getExam()